e2e/tests/upgrades/upgrade_test.go

```diff
@@ -386,6 +391,147 @@ func (s *UpgradeTestSuite) TestV5ToV6ChainUpgrade() {
 	})
 }
 
+func (s *UpgradeTestSuite) TestV6ToV7ChainUpgrade() {
```

Comment:
It might help to add a godoc asserting what we are testing. Maybe something like:

```go
// TestV6ToV7ChainUpgrade will test that an upgrade from a v6 ibc-go binary to a v7 ibc-go binary is successful
// and that the automatic migrations associated with the 02-client module are performed. Namely that the solo machine
// proto definition is migrated in state from the v2 to v3 definition. This is checked by creating a solo machine client
// before the upgrade and asserting that its TypeURL has been changed after the upgrade. The test also ensure packets
// can be sent before and after the upgrade without issue
```